schools/parse_mizs_csvs.py

```diff
@@ -61,19 +61,45 @@ def get_sledilnik_key(d, mizs_key):
 
 def reformat_dates(date_columns, row):
     """
-    Reformat the dates from a human-readable d.m.Y form into
-    the standard YMD form.
+    Reformat dates (DAT_OD, DAT_DO, DAT_POROCANJA) from a
+    human-readable d.m.Y form into the standard YMD form.
     """
+
+    # parse dates
     for i in date_columns:
         date = dateutil.parser.parse(row[i], dayfirst=True).date()
-        # fix errornous "0020" year entries
+
+        # the reporting app doen't seem to validate the year field
+        # sometimes, typos can happen
+        # fix some common ones
         if date.year == 20:
-            date = datetime(2020, date.month, date.day)
+            date = datetime(2020, date.month, date.day).date()
+        elif date.year == 2121:
+            date = datetime(2021, date.month, date.day).date()
+        elif date.year == 2201:
+            date = datetime(2021, date.month, date.day).date()
 
-        if date.year < 2020 or date.year > 2021:
-            logger.warning("Suspicious date found in line: \n{}\n".format(row))
         row[i] = date
 
+    # do some extra checks on the start and stop dates
+    start_date = row[date_columns[0]]
+    stop_date = row[date_columns[1]]
+    # start date should of course not come after the stop date
+    if start_date > stop_date:
+        logger.warning(
+            "Start date {} is after the stop date {} in line:\n{}".format(
+                start_date, stop_date, row
+            )
+        )
```

Comment:
Could be either
```python
logger.warning(f'Start date {start_date} is after the stop date {stop_date} in line:\n{row}')
```
(shorter)... or ...
```python
logger.warning('Start date %s is after the stop date %s in line:\n%s', start_date, stop_date, row)
```
to leverage filtering using logging services where you can do aggregations for this message regardless of parameter values.